src/actors/objects/mod.rs

```diff
@@ -257,131 +257,80 @@ impl CacheItemRequest for FetchFileDataRequest {
             self.0.file_id.write_sentry_scope(scope);
         });
 
-        let result = request.and_then(move |payload| {
-            if let Some(payload) = payload {
-                let download_dir =
-                    tryf!(path.parent().ok_or(ObjectErrorKind::NoTempDir)).to_owned();
-
-                let download_file = match payload {
-                    DownloadStream::FutureStream(stream) => {
-                        log::trace!("Downloading file for {}", cache_key);
-
-                        let named_download_file = tryf!(
-                            NamedTempFile::new_in(&download_dir).context(ObjectErrorKind::Io)
-                        );
-
-                        let file = tryf!(named_download_file.reopen().context(ObjectErrorKind::Io));
-
-                        let future = stream
-                            .fold(
-                                file,
-                                clone!(threadpool, |mut file, chunk| {
-                                    threadpool
-                                        .spawn_handle(async move {
-                                            file.write_all(&chunk).map(|_| file)
-                                        })
-                                        .boxed_local()
-                                        .compat()
-                                        .map_err(|_| io::Error::from(io::ErrorKind::Other))
-                                        .flatten()
-                                }),
-                            )
-                            .map(move |_| DownloadedFile::Temp(named_download_file));
-
-                        future::Either::A(future)
-                    }
-                    DownloadStream::File(path) => {
-                        log::trace!("Importing downloaded file for {}", cache_key);
-
-                        let future = fs::File::open(&path)
-                            .map(|file| DownloadedFile::Local(path.clone(), file))
-                            .context(ObjectErrorKind::Io)
-                            .map_err(ObjectError::from)
-                            .into_future();
+        let download_dir = tryf!(path.parent().ok_or(ObjectErrorKind::NoTempDir)).to_owned();
+        let download_file =
+            tryf!(NamedTempFile::new_in(&download_dir).context(ObjectErrorKind::Io));
+        let request = download_svc
+            .download(self.0.file_id.clone(), download_file.path().to_owned())
+            .map_err(Into::into);
```

Comment:
I’m a bit surprised you can `map_err` on the future here, but I guess thats a futures 0.1 thing.

Reply:
yep, futures01 are essentially a Result.